src/data/equiv/mul_add.lean

```diff
@@ -314,6 +314,29 @@ def monoid_hom_congr {M N P Q} [mul_one_class M] [mul_one_class N] [comm_monoid
   right_inv := λ k, by { ext, simp, },
   map_mul' := λ h k, by { ext, simp, }, }
 
+/-- A family of multiplicative equivalences `Π j, (Ms j ≃* Ns j)` generates a
+multiplicative equivalence between `Π j, Ms j` and `Π j, Ns j`.
+
+This is the `mul_equiv` version of `equiv.Pi_congr_right`.
+-/
+@[to_additive add_equiv.Pi_congr_right "A family of additive equivalences `Π j, (Ms j ≃+ Ns j)` generates an
+additive equivalence between `Π j, Ms j` and `Π j, Ns j`.
+
+This is the `add_equiv` version of `equiv.Pi_congr_right.`"]
+def Pi_congr_right {η : Type*}
+  {Ms Ns : η → Type*} [Π j, monoid (Ms j)] [Π j, monoid (Ns j)]
+  (es : ∀ j, Ms j ≃* Ns j) : (Π j, Ms j) ≃* (Π j, Ns j) :=
+{ map_mul' := λ x y,
+  by { ext j,
+       simp_rw [equiv.Pi_congr_right, pi.mul_apply, mul_equiv.coe_to_equiv, mul_equiv.map_mul] },
+  .. equiv.Pi_congr_right (λ j, (es j).to_equiv) }
+
+@[simp, to_additive add_equiv.Pi_congr_right_apply]
+lemma Pi_congr_right_apply {η : Type*}
+  {Ms Ns : η → Type*} [Π j, monoid (Ms j)] [Π j, monoid (Ns j)]
+  (es : ∀ j, Ms j ≃* Ns j) (x j) :
+  mul_equiv.Pi_congr_right es x j = es j (x j) := rfl
+
```

Comment:
This type of functorial definition usually should have lemmas like the following:
```suggestion

@[simp]
lemma Pi_congr_right_refl {η : Type*} {Ms : η → Type*} [Π j, mul_one_class (Ms j)] :
  Pi_congr_right (λ j, mul_equiv.refl (Ms j)) = mul_equiv.refl _ := rfl

@[simp]
lemma Pi_congr_right_symm {η : Type*}
  {Ms Ns : η → Type*} [Π j, mul_one_class (Ms j)] [Π j, mul_one_class (Ns j)]
  (es : ∀ j, Ms j ≃* Ns j) : (Pi_congr_right es).symm = (Pi_congr_right $ λ i, (es i).symm) := rfl

@[simp]
lemma Pi_congr_right_trans {η : Type*}
  {Ms Ns Ps : η → Type*} [Π j, mul_one_class (Ms j)] [Π j, mul_one_class (Ns j)]
  [Π j, mul_one_class (Ps j)]
  (es : ∀ j, Ms j ≃* Ns j) (fs : ∀ j, Ns j ≃* Ps j) :
  (Pi_congr_right es).trans (Pi_congr_right fs) = (Pi_congr_right $ λ i, (es i).trans (fs i)) := rfl
```

Reply:
Any reason for not including the `symm` version?

Reply:
`simps` already generates it (after modifying `Pi_congr_right` appropriately)

Reply:
simps generates a weaker applied version that isn't as powerful

Reply:
I thought the linter said that `simp` could prove this, but maybe I misread.

Reply:
Simp is able to prove the lemma generate by `simps` with the one I suggest, but not vice versa

Reply:
So you need `[simps apply]` to prevent it generating the offending `symm_apply` lemma.